native/src/window/id.rs

```diff
@@ -0,0 +1,16 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+/// TODO(derezzedex)
+pub struct Id(u64);
```

Comment:
Instead of hashing an arbitrary hashable as a `u64`, it might be better to define this more like [`iced_native::widget::Id`](https://docs.rs/iced_native/0.7.0/iced_native/widget/struct.Id.html)? Which is either a `usize` that is made unique by auto-incrementing, or a string.

Generating unique IDs with an incrementing integer makes sense for dynamic spawning of windows, where IDs can be stored in the model. Should be more collision resistant than hashes.